pkg/adsc/adsc.go

```diff
@@ -1133,3 +1252,55 @@ func (a *ADSC) handleMCP(gvk []string, rsc *any.Any, valBytes []byte) error {
 
 	return nil
 }
+
+// Reconnect attempts to connect, with backoff in case of failure.
+func (a *ADSC) reconnect() {
+	var err error
+	t0 := time.Now()
+	if a.adsServiceClient == nil {
+		err = a.connect()
+		log.Warna("ADSC CONNECTING ", err)
+	} else {
+		edsstr, err := a.adsServiceClient.StreamAggregatedResources(context.Background())
+		log.Warna("RESTART SERVICE ", err)
+		if err == nil {
+			a.stream = edsstr
+			// first resource in stream needs meta
+			a.sendNodeMeta = true
+		}
+	}
+
+	if err == nil && a.stream != nil {
+		a.cfg.BackoffPolicy.Reset()
+	} else {
+		time.AfterFunc(a.cfg.BackoffPolicy.NextBackOff(), a.reconnect)
```

Comment:
is this going to stack overflow after some number of reconnects?

Reply:
No, it's just scheduling a job - not recursive.